<a href="https://colab.research.google.com/github/Sanavenkatesh/Tensorflow/blob/master/celsius_to_farenheit.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

conversion from celsius to farenheit using neural network

import necessary files

In [0]:
import tensorflow as tf
import numpy as np

creating training data labels and features

In [0]:
celsius = np.array([-40, -10, 0, 8, 15, 22, 38], dtype=float)
farenheit = np.array([-40, 14, 32, 46.4, 59, 72, 100], dtype=float)

checking data

In [26]:
for i,c in enumerate(celsius):
    print("{} degrees celsius = {} degrees farenheit".format(c, farenheit[i]))

-40.0 degrees celsius = -40.0 degrees farenheit
-10.0 degrees celsius = 14.0 degrees farenheit
0.0 degrees celsius = 32.0 degrees farenheit
8.0 degrees celsius = 46.4 degrees farenheit
15.0 degrees celsius = 59.0 degrees farenheit
22.0 degrees celsius = 72.0 degrees farenheit
38.0 degrees celsius = 100.0 degrees farenheit


creating neural network layer.
syntax of creating neural network "tf.keras.layers.Dense(variables)",
here variables are units and input_shape.
units are number of nodes in the layer,
input_shape are number of inputs.



---

Here we use only one layer.

---

if we use more layers error wii be less


In [0]:
model = tf.keras.Sequential([tf.keras.layers.Dense(units=1, input_shape=[1])])

compiling the model using 'mean_squared_error' and 'Adam optimizer' 

In [0]:
model.compile(loss='mean_squared_error', optimizer=tf.keras.optimizers.Adam(0.1))

Training the model

In [29]:
history = model.fit(celsius, farenheit, epochs=500,verbose=False)
print('Finished training model')

Finished training model


predicting the values

In [30]:
print(model.predict([40.0]))

[[101.833534]]


Here we use three layers.

first layer with four nodes with one input,
second layer with four nodes (hidden layer),
third layer with one node(output layer).


In [0]:
l0 = tf.keras.layers.Dense(units=4, input_shape=[1])
l1 = tf.keras.layers.Dense(units=4)
l2 = tf.keras.layers.Dense(units=1)
model1 = tf.keras.Sequential([l0, l1, l2])


compiling and training

In [32]:
model1.compile(loss='mean_squared_error', optimizer=tf.keras.optimizers.Adam(0.1))

history = model1.fit(celsius, farenheit, epochs=500,verbose=False)
print('Finished training model')


Finished training model


prediction

In [22]:
print(model1.predict([40.0]))

[[103.93996]]


Actually 

40.0 C =104 F

with one layer we get 101.8
with three layres we get 103.9 approximately 104 F